### PR TITLE
{CI} Use MCR docker mirror

### PR DESCRIPTION
**Pull request by bebound**
Use it to prevent the docker rate limit error:
```
Unable to find image 'debian:bookworm' locally
bookworm: Pulling from library/debian
docker: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit.
```

### CODE REVIEW

**Comment by yonzhan** [comment]
CI

**Review by kairu-ms — APPROVED** [review]

File: azure-pipelines.yml [hunk 1]
```diff
@@ -323,6 +323,7 @@ jobs:
       filePath: build_scripts\windows\scripts\test_zip_installation.ps1
 
 - job: BuildDockerImageAlpine
+  continueOnError: true
```
Comment (bebound) [inline azure-pipelines.yml:326]: By adding this, we can select the partially failed CI task in release pipeline.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -547,12 +549,12 @@ jobs:
...
       echo "== Testing pip install on Python 3.9 =="
       docker run \
-        --rm -v $PYPI_FILES:/mnt/pypi python:3.9 \
+        --rm -v $PYPI_FILES:/mnt/pypi mcr.microsoft.com/mirror/docker/library/python:3.9-slim \
         /bin/bash -c "cd /mnt/pypi && ls && pip install --find-links ./ azure_cli-$CLI_VERSION*whl && az self-test && az --version && sleep 5"
       
       echo "== Testing pip install on Python 3.11 =="
       docker run \
-        --rm -v $PYPI_FILES:/mnt/pypi python:3.11 \
+        --rm -v $PYPI_FILES:/mnt/pypi mcr.microsoft.com/mirror/docker/library/python:3.11-slim \
```
Comment (jiasli) [inline azure-pipelines.yml:557]: It looks like `python:3.11` also exists and there is no need to change to `python:3.11-slim`:

https://mcr.microsoft.com/en-us/product/mirror/docker/library/python/tags

image

Comment (bebound) [reply azure-pipelines.yml:557]: `3.9` does not exists. Adding `-slim` to make sure they share the same format.

Comment (jiasli) [reply azure-pipelines.yml:557]: Well, MCR should contain `python:3.9` as it is a popular image.

**Review by jiasli — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Review by calvinhzy — APPROVED** [review]

**Comment by jiasli** [comment]
Ubuntu `noble` has been added to MCR by https://github.com/microsoft/mcr/pull/3296.

It is now available: 

https://mcr.microsoft.com/en-us/product/mirror/docker/library/ubuntu/tags

image

**Pull request merged**